### PR TITLE
Pps nano aod backport106 x

### PR DESCRIPTION
**Pull request by jwill24**
#### PR description:

This is a backport for the PPS nanoAOD tables that were added to the 11_3_X branch (PR #31531).

The main changes are:
- The naming scheme of the pps tracks "getter" functions in the ProtonProducer/ProtonFilter
- Change from directory RecoPPS -> RecoCTPPS

### CODE REVIEW

**Comment by perrotta** [comment]
backport of #31531

**Comment by perrotta** [comment]
@jwill24 
- please include also #32521 in the backport
- there are conflicts to fix (see the check report)

**Comment by perrotta** [comment]
> #### PR description:
> This is a backport for the PPS nanoAOD tables that were added to the 11_2_X branch (PR #31531).
> 

@jwill24 : the PR #31531 was  merged in 11_3_X, in fact. **Only if you need it available also in 11_2_X**, please prepare another backport for that release cycle

**Comment by silviodonato** [comment]
@jwill24 there is a conflict to be resolved. Probably another NanoAOD PR has been merged meanwhile

**Comment by jwill24** [comment]
> @jwill24 there is a conflict to be resolved. Probably another NanoAOD PR has been merged meanwhile

The conflicts have been resolved.

File: PhysicsTools/NanoAOD/plugins/BuildFile.xml [hunk 1]
```diff
@@ -17,6 +17,12 @@
 <use   name="DQMServices/Core"/>
 <use   name="CondFormats/BTauObjects"/>
 <use   name="CondTools/BTau"/>
+<use   name="DataFormats/CTPPSDetId"/>
+<use   name="DataFormats/CTPPSReco"/>
+<use   name="DataFormats/ProtonReco"/>
+<use   name="CondFormats/RunInfo"/>
+<use   name="CondFormats/DataRecord"/>
+<use   name="RecoCTPPS/ProtonReconstruction"/>
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/plugins/BuildFile.xml:25]: remove <use name="RecoPPS/ProtonReconstruction"/>
as done in master
https://github.com/cms-sw/cmssw/commit/8221dfb5c7f9892fc0646a6bb39e40b84c2a0368#diff-98594af19159a28fc4fe8d293e61a4cc09398551b79f2b2d066551ccf6cf11e5

File: PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc [hunk 2]
```diff
@@ -9,11 +9,21 @@ typedef EventSingletonSimpleFlatTableProducer<GenEventInfoProduct> SimpleGenEven
...
 #include "DataFormats/Math/interface/Point3D.h"
 typedef EventSingletonSimpleFlatTableProducer<math::XYZPointF> SimpleXYZPointFlatTableProducer;
 
 #include "FWCore/Framework/interface/MakerMacros.h"
 DEFINE_FWK_MODULE(SimpleCandidateFlatTableProducer);
 DEFINE_FWK_MODULE(SimpleGenEventFlatTableProducer);
 DEFINE_FWK_MODULE(SimpleHTXSFlatTableProducer);
+DEFINE_FWK_MODULE(SimpleProtonTrackFlatTableProducer);
+DEFINE_FWK_MODULE(SimpleLocalTrackFlatTableProducer);
 DEFINE_FWK_MODULE(SimpleXYZPointFlatTableProducer);
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc:28]: remove DEFINE_FWK_MODULE(SimpleXYZPointFlatTableProducer); 
it was added below

File: PhysicsTools/NanoAOD/python/protons_cff.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import FWCore.ParameterSet.Config as cms
+from PhysicsTools.NanoAOD.common_cff import *
+from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv1_cff import run2_nanoAOD_94XMiniAODv1
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv2_cff import run2_nanoAOD_94XMiniAODv2
+from Configuration.Eras.Modifier_run2_nanoAOD_94X2016_cff import run2_nanoAOD_94X2016
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/protons_cff.py:7]: replace all these with 

`from PhysicsTools.NanoAOD.nano_eras_cff import *`

**Review by mariadalfonso — COMMENTED** [review]
still missing additional fix from https://github.com/cms-sw/cmssw/pull/32616#issuecomment-756594133

**Comment by mariadalfonso** [comment]
please test

File: PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc (hunk 2 above)
Comment (jwill24) [reply PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc:28]: Done.

File: PhysicsTools/NanoAOD/plugins/BuildFile.xml (hunk 1 above)
Comment (jwill24) [reply PhysicsTools/NanoAOD/plugins/BuildFile.xml]: Done.

File: PhysicsTools/NanoAOD/python/protons_cff.py (hunk 3 above)
Comment (jwill24) [reply PhysicsTools/NanoAOD/python/protons_cff.py]: Done.

**Comment by perrotta** [comment]
> @jwill24
> 
> * please include also #32521 in the backport

**Comment by jwill24** [comment]
> > @jwill24
> > 
> > * please include also #32521 in the backport

@perrotta Done

**Comment by perrotta** [comment]
pelase test

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- The code for the filtered proton producer corresponds to the one merged in master (modulo different names for the PPS track getters which had been renamed between 10_6 and 11_3): this is fine with reco
- Jenkins tests pass and show no differences in reco outputs wrt the baseline, as expected: the new plugin is only used for nanoAOD productions

**Comment by mariadalfonso** [comment]
+xpog

DQM results and timing consistent with master

**Comment by qliphy** [comment]
+1

**Pull request merged**